examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -530,12 +618,11 @@ static void wfx_rsi_save_ap_info() // translation
 #endif
     if (SL_STATUS_IN_PROGRESS == status)
     {
-        const uint32_t start = osKernelGetTickCount();
-        while (!scan_results_complete && (osKernelGetTickCount() - start) <= WIFI_SCAN_TIMEOUT_TICK)
-        {
-            osThreadYield();
-        }
-        status = scan_results_complete ? callback_status : SL_STATUS_TIMEOUT;
+        StartScanTimer(WIFI_SCAN_TIMEOUT_TICK);
+
+        // Pending for the scan results pend until the scan results are received or the timer expires
+        // TODO: Remove busy wait once an event loop is implemented
+        osSemaphoreAcquire(sScanSemaphore, osWaitForever);
```

Comment:
`StartScanTimer` acquires the semaphore,  why do we acquire it here?
This seems like it will cause us to stall if the timer wasn't started.

Reply:
Because we want to pend until the timer expires or gets cancelled, we do need a way to release the semaphore so the timer can be restarted though so you have a point, will update.